### PR TITLE
Added OpenAPI generation at build time Added OpenAPI home controller to serve R4/openapi.json Updated VistaConnectionIT to also perform a read operation (which necessitated a small change to mocks)

### PR DESCRIPTION
**Pull request by bschofield-va**

### CODE REVIEW

File: vista-fhir-query/pom.xml [hunk 1]
```diff
@@ -70,6 +70,30 @@
   </dependencies>
   <build>
     <plugins>
+      <plugin>
+        <groupId>io.swagger.core.v3</groupId>
+        <artifactId>swagger-maven-plugin</artifactId>
+        <version>${swagger-maven-plugin.version}</version>
+        <executions>
+          <execution>
+            <id>r4-data-query-swag</id>
```
Comment (jhulbert-va) [inline vista-fhir-query/pom.xml:79]: copy pasta

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/api/R4VistaFhirQueryService.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            description = "US Core Implementation Guide",
+            url = "https://build.fhir.org/ig/HL7/US-Core-R4/index.html"))
+@SecurityScheme(
+    name = "OauthFlow",
+    type = SecuritySchemeType.OAUTH2,
+    in = SecuritySchemeIn.HEADER,
+    flows =
+        @OAuthFlows(
+            implicit =
+                @OAuthFlow(
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/api/R4VistaFhirQueryService.java:56]: Should there be a second oauth flow for prod?

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/observation/R4ObservationController.java [hunk 3]
```diff
@@ -187,6 +187,7 @@ public Observation read(@PathVariable("publicId") String publicId) {
     if (maybeCategory == null) {
       return null;
     }
+    //noinspection EnhancedSwitchMigration
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/observation/R4ObservationController.java:190]: ?

Comment (bschofield-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/observation/R4ObservationController.java:190]: intellij warnings ... I can't take those little yellow bars...
Usually I fix, but Fortify and CodeQL don't do well with enhanced switch statements.

**Review by jhulbert-va — APPROVED** [review]

**Review by ilaflamme-va — APPROVED** [review]

**Pull request merged**